### PR TITLE
docs: add git information docs reference and example

### PR DESCRIPTION
**Pull request by conversayShawn**

### CODE REVIEW

**Comment by MikeMcC399** [comment]
It seems that this PR is related to a Discord post https://discord.com/channels/755913899261296641/1181347411940102294 about merge queues.

Please describe in the added documentation in what situations additional environment variables are needed to be set up as when I look at the results of running [.github/workflows/example-recording.yml](https://github.com/cypress-io/github-action/blob/master/.github/workflows/example-recording.yml) sometimes the branch is correctly recorded in Cypress Cloud and sometimes not.

Does [.github/workflows/example-recording.yml](https://github.com/cypress-io/github-action/blob/master/.github/workflows/example-recording.yml) need to be updated?

**Comment by conversayShawn** [comment]
@MikeMcC399 

> If "Cloud" means "Cypress Cloud" then please change the terms to say "Cypress Cloud" explicitly.
Good call.

> Does it relate to https://github.com/cypress-io/github-action#automatic-pr-number--url-detection and should it be placed adjacent to this section?
It appears some of the info might be more readable with different structuring. Let me know if this grouping is a viable option.

> It seems that this PR is related to a Discord post https://discord.com/channels/755913899261296641/1181347411940102294 about merge queues.
Correct. The main goal here is to make it as simple as possible for users to find their answers in the documentation. 

> Does [.github/workflows/example-recording.yml](https://github.com/cypress-io/github-action/blob/master/.github/workflows/example-recording.yml) need to be updated?
Possibly, but it's not a necessity.

**Review by MikeMcC399 — CHANGES_REQUESTED** [review]
Git is written uppercase (see https://git-scm.com/)

File: README.md [hunk 1]
```diff
@@ -392,12 +392,107 @@ jobs:
...
+      - name: Cypress run
+        uses: cypress-io/github-action@v6
+        with:
+          record: true
+        env:
+          # Get the short ref name of the branch that triggered the workflow run
+          COMMIT_INFO_BRANCH: ${{ github.ref_name }}
+```
+
+Please refer to the [Cypress Cloud git information environment variables](https://docs.cypress.io/guides/continuous-integration/introduction#Git-information) section in our documentation for more examples.
```
Comment (MikeMcC399) [inline README.md:423]:
```suggestion
Please refer to the [Cypress Cloud Git information environment variables](https://on.cypress.io/guides/continuous-integration/introduction#Git-information) section in our documentation for more examples.
```

File: README.md [hunk 2]
```diff
@@ -358,21 +364,15 @@ jobs:
...
-
-**Tip 2:** we recommend passing the `GITHUB_TOKEN` secret (created by the GH Action automatically) as an environment variable. This will allow correctly identifying every build and avoid confusion when re-running a build.
-
-**Tip 3:** if running on `pull_request` event, the commit message is "merge SHA into SHA", which is not what you want probably. You can overwrite the commit message sent to Cypress Cloud by setting an environment variable. See [issue 124](https://github.com/cypress-io/github-action/issues/124#issuecomment-653180260) for details.
+### Project ID and Record Key
 
-**Tip 4:** to record the project needs `projectId`. Typically this value is saved in the [Cypress Configuration File](https://docs.cypress.io/guides/references/configuration#Configuration-File). If you want to avoid this, pass the `projectId` using an environment variable:
+To record the project needs `projectId` and `recordKey`. Typically this value is saved in the [Cypress Configuration File](https://docs.cypress.io/guides/references/configuration#Configuration-File). If you want to avoid this, pass the `projectId` using an environment variable:
```
[the remark this answers is not in the document]
Comment (MikeMcC399) [reply README.md:375]: This needs to be reworded to be correct. My suggestion to remove `recordKey` was not correct, so I deleted it.

[the remark this answers is not in the document]
Comment (conversayShawn) [reply README.md]: I can see your point here, so I reworded it to provide clarity.

**Review by jennifer-shehane — APPROVED** [review]

**Pull request merged**